s2/s2_fuzz_test.go

```diff
@@ -0,0 +1,34 @@
+package s2
```

Comment:
Please merge into `fuzz_test.go`. To be honest this should just be in `FuzzEncodingBlocks`.

Adding a Decoding fuzzer would be of much more value.